types/spectest/generate/main.go

```diff
@@ -28,18 +28,40 @@ func main() {
 		all[n] = t
 	}
 
-	byts, err := json.Marshal(all)
-	if err != nil {
-		panic(err.Error())
-	}
-
 	if len(all) != len(spectest.AllTests) {
 		panic("did not generate all tests\n")
 	}
 
-	fmt.Printf("found %d tests\n", len(all))
-	writeJson(byts)
+	// write small json files for each test
+	// try to create directory if it doesn't exist
+	_, basedir, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("no caller info")
+	}
+	testsDir := filepath.Join(strings.TrimSuffix(basedir, "main.go"), "tests")
+	if err := os.MkdirAll(testsDir, 0700); err != nil && !os.IsExist(err) {
+		panic(err.Error())
+	}
+	for name, test := range all {
+		byts, err := json.Marshal(test)
+		if err != nil {
+			panic(err.Error())
+		}
+		name = strings.ReplaceAll(name, " ", "_")
+		name = strings.ReplaceAll(name, "*", "")
+		name = "tests/" + name
+		writeJson(name, byts)
+	}
+
+	// write large tests.json file
+	byts, err := json.Marshal(all)
+	if err != nil {
+		panic(err.Error())
+	}
+	log.Printf("found %d tests\n", len(all))
```

Comment:
Can we put this on the beginning, so that it's the first print?
It becomes a bit harder to find this value due to the logs created by `writeJson(name, byts)` above.

Reply:
Thanks, updated.